returnn/vocabulary.py

```diff
@@ -36,8 +45,14 @@ def tasks(self):
     def run(self):
         lex = lexicon.Lexicon()
         lex.load(self.bliss_lexicon.get_path())
-
-        vocab = {k: v for v, k in enumerate(lex.phonemes.keys())}
+        if isinstance(self.blacklist, tk.Path):
+            blacklist = uopen(self.blacklist.get_path()).readlines()
+            blacklist = [phoneme.strip() for phoneme in blacklist]
```

Comment:
I don't like that in one case you have a set, in the other case you construct a list instead. As membership test in a set is faster, I would change this here to a set comprehension.

Not that I expect it to matter in practice as the blacklist is small.

```suggestion
            blacklist = {phoneme.strip() for phoneme in blacklist}
```